libglusterfs/src/glusterfs/common-utils.h

```diff
@@ -443,14 +444,21 @@ BIT_VALUE(unsigned char *array, unsigned int index)
         }                                                                      \
     } while (0)
 
+void gf_assert(void);
+
 #ifdef DEBUG
 #define GF_ASSERT(x) assert(x);
 #else
 #define GF_ASSERT(x)                                                           \
     do {                                                                       \
-        if (!(x)) {                                                            \
+        if (caa_unlikely(!(x))) {                                              \
+            gf_assert();                                                       \
             gf_msg_callingfn("", GF_LOG_ERROR, 0, LG_MSG_ASSERTION_FAILED,     \
-                             "Assertion failed: " #x);                         \
+                             "Assertion failed: To attach gdb and coredump,"   \
+                             " follow the command \"gdb -p $PID "              \
+                             "-ex 'set gf_signal_on_assert = 1' "              \
+                             "-ex 'handle SIGCONT stop pass' "                 \
+                             "-ex 'cont' -ex 'gcore' -ex 'quit'\"");           \
```

Comment:
I'm not sure if it's a good idea to write this in the log message. But if we write it, we should point to an script to do that, not a raw command. You can look at the issue. I proposed a script to do all the work.

Reply:
Agree. I was not aware of the script before submitting the patch. I will add the script to this patch and make the correction to the msg. Where can it be accommodated? "/extras/debug/<script>"?